proposals/0229-simd.md

```diff
@@ -226,22 +226,22 @@ public extension SIMDVector {
   init<S: Sequence>(_ elements: S) where S.Element == Scalar
 }
 
-public extension SIMDVector where Scalar : Comparable {
+public extension SIMD where Scalar : Comparable {
   /// Pointwise ordered comparisons
-  static func .<(lhs: Self, rhs: Self) -> Mask
-  static func .<=(lhs: Self, rhs: Self) -> Mask 
-  static func .>=(lhs: Self, rhs: Self) -> Mask
-  static func .>(lhs: Self, rhs: Self) -> Mask 
+  static func .<(lhs: Self, rhs: Self) -> SIMDMask<MaskStorage>
+  static func .<=(lhs: Self, rhs: Self) -> SIMDMask<MaskStorage> 
+  static func .>=(lhs: Self, rhs: Self) -> SIMDMask<MaskStorage>
+  static func .>(lhs: Self, rhs: Self) -> SIMDMask<MaskStorage> 
   
-  static func .<(lhs: Scalar, rhs: Self) -> Mask
-  static func .<=(lhs: Scalar, rhs: Self) -> Mask 
-  static func .>=(lhs: Scalar, rhs: Self) -> Mask
-  static func .>(lhs: Scalar, rhs: Self) -> Mask 
+  static func .<(lhs: Scalar, rhs: Self) -> SIMDMask<MaskStorage>
+  static func .<=(lhs: Scalar, rhs: Self) -> SIMDMask<MaskStorage> 
+  static func .>=(lhs: Scalar, rhs: Self) -> SIMDMask<MaskStorage>
+  static func .>(lhs: Scalar, rhs: Self) -> SIMDMask<MaskStorage> 
   
-  static func .<(lhs: Self, rhs: Scalar) -> Mask
-  static func .<=(lhs: Self, rhs: Scalar) -> Mask 
-  static func .>=(lhs: Self, rhs: Scalar) -> Mask
-  static func .>(lhs: Self, rhs: Scalar) -> Mask 
+  static func .<(lhs: Self, rhs: Scalar) -> SIMDMask<MaskStorage>
+  static func .<=(lhs: Self, rhs: Scalar) -> SIMDMask<MaskStorage> 
+  static func .>=(lhs: Self, rhs: Scalar) -> SIMDMask<MaskStorage>
+  static func .>(lhs: Self, rhs: Scalar) -> SIMDMask<MaskStorage> 
 }
 
 public extension SIMDMaskVector {
```

Comment:
```suggestion
public extension SIMDMask {
```

(and change `Self` to `SIMDMask` below).

Reply:
The current implementation has `Bool` parameters (instead of `Scalar`).

```swift
public extension SIMDMask {
  /// Boolean operations on Masks; these use the .-prefixed form because
  /// they do not short-circuit like `&&` and `||`, but the normal
  /// boolean operators have the wrong precedence for the typical use
  /// of these operators.
  static prefix func .!(rhs: SIMDMask) -> SIMDMask
  
  static func .&(lhs: SIMDMask, rhs: SIMDMask) -> SIMDMask
  static func .^(lhs: SIMDMask, rhs: SIMDMask) -> SIMDMask
  static func .|(lhs: SIMDMask, rhs: SIMDMask) -> SIMDMask
  
  static func .&(lhs: Bool, rhs: SIMDMask) -> SIMDMask
  static func .^(lhs: Bool, rhs: SIMDMask) -> SIMDMask
  static func .|(lhs: Bool, rhs: SIMDMask) -> SIMDMask
  
  static func .&(lhs: SIMDMask, rhs: Bool) -> SIMDMask
  static func .^(lhs: SIMDMask, rhs: Bool) -> SIMDMask
  static func .|(lhs: SIMDMask, rhs: Bool) -> SIMDMask
  
  static func .&=(lhs: inout SIMDMask, rhs: SIMDMask)
  static func .^=(lhs: inout SIMDMask, rhs: SIMDMask)
  static func .|=(lhs: inout SIMDMask, rhs: SIMDMask)
  
  static func .&=(lhs: inout SIMDMask, rhs: Bool)
  static func .^=(lhs: inout SIMDMask, rhs: Bool)
  static func .|=(lhs: inout SIMDMask, rhs: Bool)
  
  static func random<T: RandomNumberGenerator>(using generator: inout T) -> SIMDMask
  static func random() -> SIMDMask
}
```

Do you need to include any of the other `SIMDMask` APIs?

```swift
public struct SIMDMask<Storage> : SIMD
                  where Storage : SIMD,
                 Storage.Scalar : FixedWidthInteger & SignedInteger {
  
  public var _storage : Storage
  
  public typealias MaskStorage = Storage
  
  public typealias Scalar = Bool
  
  public var scalarCount: Int { get }
  
  public init()
  
  public init(_ _storage: Storage)
  
  public subscript(index: Int) -> Bool { get set }
}
```

Reply:
Updated Scalar -> Bool. The other stuff is just the `SIMD` requirements (and the one `_storage` init, which is an implementation detail), so already included above.